### PR TITLE
Use opcode byte instead of int 

### PR DESCRIPTION
**Pull request by 1Jo1**
Motivation:

we should use byte instead of int as the size of opcode is 1 byte

Modification:

use byte instead of int

Result:

we have more space in user_data

### CODE REVIEW

**Review by franz1981 — APPROVED** [review]

**Comment by normanmaurer** [comment]
@netty-bot test this please

File: transport-native-io_uring/src/main/java/io/netty/channel/uring/UserData.java [hunk 1]
```diff
@@ -19,15 +19,13 @@
...
+    static long encode(int fd, byte op, short data) {
+        return ((long) data << 48) | ((op & 0xFFL)  << 32) | fd & 0xFFFFFFFFL;
     }
 
     static void decode(int res, int flags, long udata, IOUringCompletionQueueCallback callback) {
         int fd = (int) (udata & 0xFFFFFFFFL);
-        int op = (short) ((udata >>>= 32) & 0xFFFFL);
+        int op = (short) ((udata >>>= 32) & 0xFFL);
         short data = (short) (udata >>> 16);
         callback.handle(fd, res, flags, op, data);
```
Comment (normanmaurer) [inline transport-native-io_uring/src/main/java/io/netty/channel/uring/UserData.java:30]: @1Jo1 we should change the callback handle method to also use byte for the op.

**Review by normanmaurer — CHANGES_REQUESTED** [review]

**Review by njhill — APPROVED** [review]
Same comment as @normanmaurer

**Comment by normanmaurer** [comment]
@netty-bot test this please

**Comment by normanmaurer** [comment]
I did make the callback fix.

**Pull request merged**